content/docs/reference/design-docs/conflict-detection/index.en.md

```diff
@@ -0,0 +1,139 @@
+to become impossible to perform.
+
+Conflict detection relies on interlocking and signaling simulation to:
+1) figure out what each actor requires to perform its duty undisturbed
+2) detect conflicting requirements
+
+This system has to:
+ - produce conflicts which can be linked back to a root cause
+ - operate in way that can be visualized and interpreted
+ - be scalable: it should work just as fast given a massive number of train paths to intersect
+ - enable threading new train paths into an existing timetable
+
+The design of this system is guided by a number of constraints:
+ - **it must not cause false-negatives**: if no conflicts are detected, a multi-train simulation of the same timetable must no yield any slowdowns
+ - it cannot rely on data we do not have
+ - it has to enable later support of mobile block systems
+ - it has to rely on existing signaling and interlocking simulation
+ - it has to enable detecting conflicts regardless of the signaling system in use
+ - support transitions between signaling systems
+ - support conflicts between different signaling systems
+
+## Conflict theory
+
+Conflict detection relies on an abstract model:
+ - objects where conflicts happen are resources
+ - resources are used by actors
+ - resource uses can be incompatible
+ - overlapping incompatible ressource uses generate a conflict
+
+### Actors
+
+Actors are objects which cause resources to be used:
+ - **train paths** (or someone / something on the behalf of the train)
+ - **planned infrastructure work**
+
+### Resources
+
+Resources have the following properties:
+ - can be used by actors
+ - have multiple states
+ - can only be in a single state at any given time
+ - may take time changing state
+
+The current model only considers the following two resources:
+ - **zones**, which have one state per way to traverse it
+ - **switches**, which have one state per position
+
+{{% pageinfo color="info" %}}
+Station platforms could also be resources, which would enable modeling platform use conflicts: if two tracks share a tiny platform, it may not be allowed to use both sides of the platform at once.
+{{% /pageinfo %}}
+
+### Requirements and conflicts
+
+A **requirement** describes what resource is needed by an actor, and when.
+
+There are a few types of requirements:
+ - **shareable**: the resource is required to be in a given configuration, for a given time lapse. Other shareable requirements with the same configuration can stack up.
+ - **exclusive**: the resource is required to be in a given configuration, for a given actor, for a given time lapse. This requirement cannot be combined with others.
+
+**Conflicts** are both real-life disturbances and the incompatible resource requirements which cause them. A disturbance is something that would cause a train to be unable to run as planned.
+ - route setup delays due to unavailable resources
+ - spacing conflicts due to a train catching up with another
```

Comment:
There can also be convergence conflicts, or nose-to-nose, etc that are not spacing conflict striclty speaking

Reply:
Not quite sure of the nose-to-nose english expression :D

Reply:
yeah it's head to head